### PR TITLE
chore: fuzz test expansion

### PR DESCRIPTION
**Pull request by romanzac**
## Description

Add fuzz tests for edge cases to improve unit tests coverage.

## Tests added

- testFuzz_RegisterInvalid

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [] Added natspec comments?
- [] Ran `pnpm adorno`?

### CODE REVIEW

File: test/WakuRlnV2.fuzz.t.sol [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+                initialTotal += rateLimit;
+            }
+        }
+
+        // Fuzz time warp
+        uint256 minDelta =
+            uint256(w.activeDurationForNewMemberships()) + uint256(w.gracePeriodDurationForNewMemberships()) + 1;
+        vm.warp(block.timestamp + minDelta);
+
+        uint256[] memory idCommitments = _buildIdsFromMask(subsetMask);
```
Comment (gravityblast) [inline test/WakuRlnV2.fuzz.t.sol:114]: don't we want to ensure that the length of idCommitments is at least 1?

Comment (romanzac) [reply test/WakuRlnV2.fuzz.t.sol:114]: Length of idCommitments cannot be 0. This is guaranteed by subsetMask > 0 in`vm.assume(subsetMask > 0 && subsetMask < 16)`. I can add `assert(idCommitments.length >= 1)` if you want ? Here is the complete output set:

subsetMask (dec) | subsetMask (bin) | idCommitments
1 | 0001 | [1]
2 | 0010 | [2]
3 | 0011 | [1,2]
4 | 0100 | [3]
5 | 0101 | [1,3]
6 | 0110 | [2,3]
7 | 0111 | [1,2,3]
8 | 1000 | [4]
9 | 1001 | [1,4]
10 | 1010 | [2,4]
11 | 1011 | [1,2,4]
12 | 1100 | [3,4]
13 | 1101 | [1,3,4]
14 | 1110 | [2,3,4]
15 | 1111 | [1,2,3,4]

Comment (gravityblast) [reply test/WakuRlnV2.fuzz.t.sol:114]: oh yeah you are right, I missed the `subsetMask < 16 `
